### ISSUE TITLE
Crash upon startup of v2.4.11

### ISSUE THREAD

**Issue by martin5233** [traefik/traefik#8295]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

After upgrading from Traefik 2.4.9 to 2.4.11 Traefik immediately crashes after startup. The output is as attached. The problem must have been introduced with 2.4.11 or 2.4.10 (which isn't available on Docker Hub), the same configuration worked flawlessly with 2.4.9

### What did you see instead?

Crash upon startup

### What version of Traefik are you using?

Version:      2.4.11
Codename:     livarot
Go version:   go1.16.6
Built:        2021-07-15T15:03:36Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

Traefik is run as a docker container with a rather minimal docker-compose file:

```yml
version : '3'
services:
  traefik:
    build:
      context: .
      args:
        certname: "${CERTNAME}"
    networks:
      - default
    ports:
      - "9090:9090"       
      - "9000:9000"    
      - "27777:27777"  
      - "9002:9002"    
      - "80:80"      
      - "443:443"    
      - "1080:1080"     
      - "8080:8080"    
      - "27001:27001"  
      - "33133:33133"  
      - "3307:3307"    
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    container_name: "traefik"
    restart: unless-stopped
    labels:
      - "com.docker.compose.project=traefik"
      - "traefik.http.routers.api.rule=Host(`${FQDN_NAME}`)"
      - "traefik.http.routers.api.entrypoints=traefik"
      - "traefik.http.routers.api.service=api@internal"
      - "traefik.http.routers.api.tls"
      - "traefik.enable=true"
networks:
  default:
    external:
      name: traefik
```

and the underlying Dockerfile as follows:
```dockerfile
FROM traefik:v2.4
ARG certname
RUN mkdir -p /etc/traefik/dynamic/certs
COPY traefik.toml /etc/traefik/traefik.toml
COPY dynamic /etc/traefik/dynamic
# Replace certname placeholder by actual certname, so container uses the correct certificate
RUN sed -e s/\${CERTNAME}/$certname/ /etc/traefik/dynamic/tls.toml.in >/etc/traefik/dynamic/tls.toml
RUN rm /etc/traefik/dynamic/tls.toml.in
```
 
traefik.toml: 
```toml
[global]
  checkNewVersion = true
  sendAnonymousUsage = true

################################################################
# Entrypoints configuration
################################################################

[entryPoints]
  [entryPoints.traefik]
    address = ":9090"

  [entryPoints.jenkins-web]
    address = ":9000"

  [entryPoints.jenkins-ssh]
    address = ":9002"

  [entryPoints.jenkins-node]
    address = ":27777"

  [entryPoints.web]
    address = ":80"       

  [entryPoints.webssl]
    address = ":443"    

  [entryPoints.davmail]
    address = ":1080"

  [entryPoints.klocwork-web]
    address = ":8080"

  [entryPoints.klocwork-license]
    address = ":27001"

  [entryPoints.klocwork-lmgrd]
    address = ":33133"

  [entryPoints.klocwork-db]
    address = ":3307"

################################################################
# Traefik logs configuration
################################################################

[log]

  level = "DEBUG"

# Enable API and dashboard
[api]

  insecure = false

  dashboard = true

# Enable ping
[ping]

[providers.docker]

[ping]

# Enable Docker configuration backend
[providers.docker]
  exposedByDefault = false
  network = "traefik"

[providers.file]
   directory = "/etc/traefik/dynamic"

```

### If applicable, please paste the log output in DEBUG level

```
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Configuration loaded from file: /etc/traefik/traefik.toml"
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Traefik version 2.4.11 built on 2021-07-15T15:03:36Z"
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Static configuration loaded {\"global\":{\"checkNewVersion\":true,\"sendAnonymousUsage\":true},\"serversTransport\":{\"maxIdleConnsPerHost\":200},\"entryPoints\":{\"davmail\":{\"address\":\":1080\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"jenkins-node\":{\"address\":\":27777\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"jenkins-ssh\":{\"address\":\":9002\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"jenkins-web\":{\"address\":\":9000\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"klocwork-db\":{\"address\":\":3307\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"klocwork-license\":{\"address\":\":27001\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"klocwork-lmgrd\":{\"address\":\":33133\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":
{},\"http\":{}},\"klocwork-web\":{\"address\":\":8080\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\"3m0s\"}},\"f
orwardedHeaders\":{},\"http\":{}},\"traefik\":{\"address\":\":9090\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTimeout\":\
"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"web\":{\"address\":\":80\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":{\"idleTime
out\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}},\"webssl\":{\"address\":\":443\",\"transport\":{\"lifeCycle\":{\"graceTimeOut\":\"10s\"},\"respondingTimeouts\":
{\"idleTimeout\":\"3m0s\"}},\"forwardedHeaders\":{},\"http\":{}}},\"providers\":{\"providersThrottleDuration\":\"2s\",\"docker\":{\"watch\":true,\"endpoint\":\"unix:/
//var/run/docker.sock\",\"defaultRule\":\"Host(`{{ normalize .Name }}`)\",\"network\":\"traefik\",\"swarmModeRefreshSeconds\":\"15s\"},\"file\":{\"directory\":\"/etc/
traefik/dynamic\",\"watch\":true}},\"api\":{\"dashboard\":true},\"ping\":{\"entryPoint\":\"traefik\",\"terminatingStatusCode\":503},\"log\":{\"level\":\"DEBUG\",\"for
mat\":\"common\"},\"pilot\":{\"dashboard\":true}}"
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Stats collection is enabled."
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Many thanks for contributing to Traefik's improvement by allowing us to receive anonymous information from yo
ur configuration."
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Help us improve Traefik by leaving this feature on :)"
traefik    | time="2021-07-23T08:32:03Z" level=info msg="More details on: https://doc.traefik.io/traefik/contributing/data-collection/"
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=klocwork-db
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=jenkins-web
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=jenkins-node
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=davmail
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=web
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Starting provider aggregator.ProviderAggregator {}"
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=klocwork-license
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=klocwork-web
traefik    | time="2021-07-23T08:32:03Z" level=info msg="Starting provider *file.Provider {\"directory\":\"/etc/traefik/dynamic\",\"watch\":true}"
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=jenkins-ssh
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=webssl
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=klocwork-lmgrd
traefik    | time="2021-07-23T08:32:03Z" level=debug msg="Start TCP Server" entryPointName=traefik
traefik    | time="2021-07-23T08:32:03Z" level=error msg="Error in Go routine: runtime error: invalid memory address or nil pointer dereference"
traefik    | time="2021-07-23T08:32:03Z" level=error msg="Stack: goroutine 81 [running]:\nruntime/debug.Stack(0xc00017e000, 0x343c2ba, 0x17)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9f\ngithub.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x2e89ec0, 0x546faf0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:66 +0xb4\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1.1(0x3557b00)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:56 +0x5a\npanic(0x2e89ec0, 0x546faf0)\n\t/usr/local/go/src/runtime/panic.go:965 +0x1b9\ngithub.com/traefik/traefik/v2/pkg/provider/file.(*Provider).loadFileConfig(0xc0004efa40, 0x3999d80, 0xc0004ab0b0, 0xc000220c80, 0x1d, 0x3999d01, 0xc000764660, 0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:194 +0xb20\ngithub.com/traefik/traefik/v2/pkg/provider/file.(*Provider).loadFileConfigFromDirectory(0xc0004efa40, 0x3999d80, 0xc0004ab0b0, 0xc0004f6f60, 0x14, 0xc000142160, 0xc0004ab0b0, 0x1, 0xc00075dd38)\n\t/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:328 +0x599\ngithub.com/traefik/traefik/v2/pkg/provider/file.(*Provider).BuildConfiguration(0xc0004efa40, 0xc0009225e0, 0x2, 0x2)\n\t/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:81 +0x197\ngithub.com/traefik/traefik/v2/pkg/provider/file.(*Provider).Provide(0xc0004efa40, 0xc0002e7920, 0xc00081f2f0, 0xc0009225e0, 0x2)\n\t/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:49 +0x32\ngithub.com/traefik/traefik/v2/pkg/provider/aggregator.launchProvider(0xc0002e7920, 0xc00081f2f0, 0x3959690, 0xc0004efa40)\n\t/go/src/github.com/traefik/traefik/pkg/provider/aggregator/aggregator.go:151 +0x1c2\ngithub.com/traefik/traefik/v2/pkg/provider/aggregator.ProviderAggregator.Provide(0x3959de8, 0xc000340850, 0x3959690, 0xc0004efa40, 0xc00020d7a0, 0x2, 0x2, 0xc0002e7920, 0xc00081f2f0, 0xc00093e420, ...)\n\t/go/src/github.com/traefik/traefik/pkg/provider/aggregator/aggregator.go:123 +0x1bb\ngithub.com/traefik/traefik/v2/pkg/server.(*ConfigurationWatcher).startProvider.func1()\n\t/go/src/github.com/traefik/traefik/pkg/server/configurationwatcher.go:96 +0x68\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1(0x3557b00, 0xc000630ba0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:59 +0x4f\ncreated by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:53 +0x49\n"
```

```
Stack: goroutine 81 [running]:
runtime/debug.Stack(0xc00017e000, 0x343c2ba, 0x17)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x2e89ec0, 0x546faf0)
	/go/src/github.com/traefik/traefik/pkg/safe/routine.go:66 +0xb4
github.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1.1(0x3557b00)
	/go/src/github.com/traefik/traefik/pkg/safe/routine.go:56 +0x5a
panic(0x2e89ec0, 0x546faf0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/traefik/traefik/v2/pkg/provider/file.(*Provider).loadFileConfig(0xc0004efa40, 0x3999d80, 0xc0004ab0b0, 0xc000220c80, 0x1d, 0x3999d01, 0xc000764660, 0x0, 0x0)
	/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:194 +0xb20
github.com/traefik/traefik/v2/pkg/provider/file.(*Provider).loadFileConfigFromDirectory(0xc0004efa40, 0x3999d80, 0xc0004ab0b0, 0xc0004f6f60, 0x14, 0xc000142160, 0xc0004ab0b0, 0x1, 0xc00075dd38)
	/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:328 +0x599
github.com/traefik/traefik/v2/pkg/provider/file.(*Provider).BuildConfiguration(0xc0004efa40, 0xc0009225e0, 0x2, 0x2)
	/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:81 +0x197
github.com/traefik/traefik/v2/pkg/provider/file.(*Provider).Provide(0xc0004efa40, 0xc0002e7920, 0xc00081f2f0, 0xc0009225e0, 0x2)
	/go/src/github.com/traefik/traefik/pkg/provider/file/file.go:49 +0x32
github.com/traefik/traefik/v2/pkg/provider/aggregator.launchProvider(0xc0002e7920, 0xc00081f2f0, 0x3959690, 0xc0004efa40)
	/go/src/github.com/traefik/traefik/pkg/provider/aggregator/aggregator.go:151 +0x1c2
github.com/traefik/traefik/v2/pkg/provider/aggregator.ProviderAggregator.Provide(0x3959de8, 0xc000340850, 0x3959690, 0xc0004efa40, 0xc00020d7a0, 0x2, 0x2, 0xc0002e7920, 0xc00081f2f0, 0xc00093e420, ...)
	/go/src/github.com/traefik/traefik/pkg/provider/aggregator/aggregator.go:123 +0x1bb
github.com/traefik/traefik/v2/pkg/server.(*ConfigurationWatcher).startProvider.func1()
	/go/src/github.com/traefik/traefik/pkg/server/configurationwatcher.go:96 +0x68
github.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1(0x3557b00, 0xc000630ba0)
	/go/src/github.com/traefik/traefik/pkg/safe/routine.go:59 +0x4f
created by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover
	/go/src/github.com/traefik/traefik/pkg/safe/routine.go:53 +0x49
```

**Comment by ldez** [comment]
Hello,

Can you provide your `tls.toml` file?

**Comment by martin5233** [comment]
Hi Idez, 

sure, here it is:

```
[tls.stores]
  [tls.stores.default]

[[tls.certificates]]
  certFile = "/etc/traefik/dynamic/certs/${CERTNAME}.pem"
  keyFile = "/etc/traefik/dynamic/certs/${CERTNAME}.key"
  stores = ["default"]
```

**Comment by ldez** [comment]
you don't need to create and set the `default` store:

```toml
[[tls.certificates]]
  certFile = "/etc/traefik/dynamic/certs/${CERTNAME}.pem"
  keyFile = "/etc/traefik/dynamic/certs/${CERTNAME}.key"
```

**Comment by martin5233** [comment]
You are right, without the default store, the crash does not occur. I will remove it from our configuration, but obviously Traefik should not crash, if this is present ;-)

**Comment by ldez** [comment]
yes we will fix that, thank you for your report :+1:

**Comment by martin5233** [comment]
Thank you for your very quick response!

**Comment by martin5233** [comment]
I probably shouldn't have closed this.

**Comment by ldez** [comment]
In fact, it's a duplicate of #8273

It has been fixed in #8274, it will be a part of the next release (v2.4.12)
